packages/predicate/src/utils/getContractRoot.ts

```diff
@@ -13,5 +14,21 @@ export const getContractRoot = (bytecode: BytesLike): string => {
     chunks.push(chunk);
   }
 
-  return calcRoot(chunks.map((c) => hexlify(c)));
+  const totalBytes = chunks.reduce((sum, chunk) => chunk.byteLength + sum, 0);
+  const lastChunk = chunks[chunks.length - 1];
+  const isDivisibleBy16 = totalBytes % 16 === 0;
+  const remainingBytes = chunkSize - lastChunk.length;
+  if (!isDivisibleBy16) {
+    const nearestMultiple = Math.ceil(remainingBytes / 8) * 8;
+    const paddedChunkLength = lastChunk.length + nearestMultiple;
+    const paddedChunk = new Uint8Array(paddedChunkLength).fill(0);
+    paddedChunk.set(lastChunk, 0);
+    chunks[chunks.length - 1] = paddedChunk;
+  }
+
+  const chainIdBytes = uint64ToBytesBE(chainId);
+  const codeRoot = calcRoot(chunks.map((c) => hexlify(c)));
+
+  const contractRoot = hash(concat(['0x4655454C', chainIdBytes, codeRoot]));
+  return contractRoot;
```

Comment:
The spec for how we calculate the predicate's ID/address has changed: https://github.com/FuelLabs/fuel-specs/blob/a937dac06f6eda0905c2e3bcbee6038d39748b32/src/protocol/id/predicate.md